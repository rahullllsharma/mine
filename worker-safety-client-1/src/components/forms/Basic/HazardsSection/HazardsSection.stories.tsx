import type { ComponentMeta, ComponentStory } from "@storybook/react";

import { HazardsSection } from "./HazardsSection";

export default {
  title: "JSB/HazardsSection",
  component: HazardsSection,
} as ComponentMeta<typeof HazardsSection>;

export const Playground: ComponentStory<typeof HazardsSection> = args => (
  <HazardsSection {...args} />
);

Playground.args = {
  onClickEdit: alert,
  hazards: [
    "Stuck by equipment",
    "Electrical contact with source",
    "Biological hazards - Insects, animals, poisonous plants",
  ],
};
