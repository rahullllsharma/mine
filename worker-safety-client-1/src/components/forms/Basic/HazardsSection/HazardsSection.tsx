import { BodyText } from "@urbint/silica";
import { GroupDiscussionSection } from "../GroupDiscussionSection";

export type HazardsSectionData = {
  hazards: Array<string>;
};

export type HazardsSectionProps = HazardsSectionData & {
  onClickEdit: () => void;
};

const HazardsSection = ({ onClickEdit, hazards }: HazardsSectionProps) => {
  return (
    <GroupDiscussionSection title="Hazards" onClickEdit={onClickEdit}>
      <div className="flex flex-col gap-1">
        {hazards.map(hazard => (
          <div key={hazard} className="bg-white p-4 shadow-10">
            <BodyText className="text-base font-semibold text-neutral-shade-75">
              {hazard}
            </BodyText>
          </div>
        ))}
      </div>
    </GroupDiscussionSection>
  );
};

export { HazardsSection };
