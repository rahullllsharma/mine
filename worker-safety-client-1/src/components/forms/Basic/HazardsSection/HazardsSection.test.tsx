import { render, screen, fireEvent } from "@testing-library/react";

import { HazardsSection } from "./HazardsSection";

describe(HazardsSection.name, () => {
  const props = {
    onClickEdit: jest.fn(),
    hazards: [
      "Stuck by equipment",
      "Electrical contact with source",
      "Biological hazards - Insects, animals, poisonous plants",
    ],
  };
  it("renders correctly", () => {
    const { asFragment } = render(<HazardsSection {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });

  describe("user events", () => {
    it("does correct action when user clicks on edit button", () => {
      const onClickEdit = jest.fn();
      render(<HazardsSection {...props} onClickEdit={onClickEdit} />);

      fireEvent.click(screen.getByRole("button"));

      expect(onClickEdit).toHaveBeenCalledTimes(1);
    });
  });
});
