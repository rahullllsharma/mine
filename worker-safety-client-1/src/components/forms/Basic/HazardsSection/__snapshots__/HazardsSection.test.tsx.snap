// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HazardsSection renders correctly 1`] = `
<DocumentFragment>
  <fieldset
    class="flex flex-col gap-3"
  >
    <div
      class="p-2 sm:p-4 flex flex-col gap-3 bg-brand-gray-10"
    >
      <div
        class="flex justify-between"
      >
        <h2
          class="text-section-heading font-section-heading text-neutrals-primary text-lg font-semibold text-neutral-shade-75"
        >
          Hazards
        </h2>
        <button
          class="text-brand-urbint-50 flex items-center"
        >
          <i
            aria-hidden="true"
            class="ci-edit"
          />
          <span
            class="ml-1"
          >
            Edit
          </span>
        </button>
      </div>
      <div
        class="mt-4"
      >
        <div
          class="flex flex-col gap-1"
        >
          <div
            class="bg-white p-4 shadow-10"
          >
            <p
              class="text-body-text font-body-text text-neutrals-primary text-base font-semibold text-neutral-shade-75"
            >
              Stuck by equipment
            </p>
          </div>
          <div
            class="bg-white p-4 shadow-10"
          >
            <p
              class="text-body-text font-body-text text-neutrals-primary text-base font-semibold text-neutral-shade-75"
            >
              Electrical contact with source
            </p>
          </div>
          <div
            class="bg-white p-4 shadow-10"
          >
            <p
              class="text-body-text font-body-text text-neutrals-primary text-base font-semibold text-neutral-shade-75"
            >
              Biological hazards - Insects, animals, poisonous plants
            </p>
          </div>
        </div>
      </div>
    </div>
  </fieldset>
</DocumentFragment>
`;
